app/src/main/java/com/github/onlynotesswent/ui/search/Search.kt

```diff
@@ -210,45 +287,108 @@ fun SearchScreen(
         val displayUsers: Boolean =
             searchType.value == SearchType.USERS && filteredUsers.value.isNotEmpty()
 
-        val displayNotes: Boolean =
-            searchType.value == SearchType.NOTES && filteredNotes.value.isNotEmpty()
+        val displayPublicNotes: Boolean =
+            searchType.value == SearchType.NOTES &&
+                additionalFilter.value == AdditionalFilterType.PUBLIC &&
+                filteredPublicNotes.value.isNotEmpty()
+
+        val displayFriendsNotes: Boolean =
+            searchType.value == SearchType.NOTES &&
+                additionalFilter.value == AdditionalFilterType.FRIENDS &&
+                filteredFriendsNotes.value.isNotEmpty()
+
+        val displayPublicFolders: Boolean =
+            searchType.value == SearchType.FOLDERS &&
+                additionalFilter.value == AdditionalFilterType.PUBLIC &&
+                filteredPublicFolders.value.isNotEmpty()
+
+        val displayFriendsFolders: Boolean =
+            searchType.value == SearchType.FOLDERS &&
+                additionalFilter.value == AdditionalFilterType.FRIENDS &&
+                filteredFriendsFolders.value.isNotEmpty()
 
-        val displayFolders: Boolean =
-            searchType.value == SearchType.FOLDERS && filteredFolders.value.isNotEmpty()
+        val displayPublicDecks: Boolean =
+            searchType.value == SearchType.DECKS &&
+                additionalFilter.value == AdditionalFilterType.PUBLIC &&
+                filteredPublicDecks.value.isNotEmpty()
 
-        val displayDecks: Boolean =
-            searchType.value == SearchType.DECKS && filteredDecks.value.isNotEmpty()
+        val displayFriendsDecks: Boolean =
+            searchType.value == SearchType.DECKS &&
+                additionalFilter.value == AdditionalFilterType.FRIENDS &&
+                filteredFriendsDecks.value.isNotEmpty()
 
         val displayLoader: Boolean =
```

Comment:
Since at most one of the display values can be true at a time, we could define an enum or assign an index to each boolean. Since the search will proably not change a lot right now, you can leave it like this temporarily.
Moreover, the only case where they are all false is when the search query is empty and the search type returns no values. For this reason, `displayLoader` could be simplified by:
(`searchQuery` is not empty) and (all the other display booleans are false)

Reply:
I agree, I refactored it to have a more concise logic